### PR TITLE
VM: add selectHardforkByBlockNumber option

### PR DESCRIPTION
**Pull request by jochem-brouwer**
This PR adds an option to make the VM automatically switch to the right hardfork in `runBlock`.

### CODE REVIEW

File: packages/vm/lib/index.ts [hunk 1]
```diff
@@ -89,6 +89,14 @@ export interface VMOpts {
    * Default: `false` [ONLY set to `true` during debugging]
    */
   allowUnlimitedContractSize?: boolean
+
+  /**
+   * Select hardfork based upon block number. This automatically switches to the right hard fork based upon the block number.
+   *
+   * Default: `true`
+   */
+
```
Comment (ryanio) [inline packages/vm/lib/index.ts:98]: nit: prefer no newline here to match above
```suggestion
```

**Review by ryanio — APPROVED** [review]
looks good, nice test case!

**Comment by ryanio** [comment]
next time I will try adding `type: test all hardforks` label before the commit, didn't run with only a re-run

File: packages/vm/lib/index.ts [hunk 2]
```diff
@@ -89,6 +89,13 @@ export interface VMOpts {
    * Default: `false` [ONLY set to `true` during debugging]
    */
   allowUnlimitedContractSize?: boolean
+
+  /**
+   * Select hardfork based upon block number. This automatically switches to the right hard fork based upon the block number.
+   *
+   * Default: `true`
+   */
+  selectHardforkByBlockNumber?: boolean
```
Comment (holgerd77) [inline packages/vm/lib/index.ts:98]: ???

We can't having this default to `true`, this is a giga-breaking change.

Comment (jochem-brouwer) [reply packages/vm/lib/index.ts:98]: Yes you are right, will default this to `false` in a new PR.

**Pull request merged**